stdlib/socket.pyi

```diff
@@ -1,509 +1,385 @@
-    TIPC_SRC_DROPPABLE: int
-    TIPC_SUBSCR_TIMEOUT: int
-    TIPC_SUB_CANCEL: int
-    TIPC_SUB_PORTS: int
-    TIPC_SUB_SERVICE: int
-    TIPC_TOP_SRV: int
-    TIPC_WAIT_FOREVER: int
-    TIPC_WITHDRAWN: int
-    TIPC_ZONE_SCOPE: int
-
-if sys.platform == "linux":
-    AF_ALG: AddressFamily
-    SOL_ALG: int
-    ALG_OP_DECRYPT: int
-    ALG_OP_ENCRYPT: int
-    ALG_OP_SIGN: int
-    ALG_OP_VERIFY: int
-    ALG_SET_AEAD_ASSOCLEN: int
-    ALG_SET_AEAD_AUTHSIZE: int
-    ALG_SET_IV: int
-    ALG_SET_KEY: int
-    ALG_SET_OP: int
-    ALG_SET_PUBKEY: int
-
-if sys.platform == "linux" and sys.version_info >= (3, 7):
-    AF_VSOCK: AddressFamily
-    IOCTL_VM_SOCKETS_GET_LOCAL_CID: int
-    VMADDR_CID_ANY: int
-    VMADDR_CID_HOST: int
-    VMADDR_PORT_ANY: int
-    SO_VM_SOCKETS_BUFFER_MAX_SIZE: int
-    SO_VM_SOCKETS_BUFFER_SIZE: int
-    SO_VM_SOCKETS_BUFFER_MIN_SIZE: int
-    VM_SOCKETS_INVALID_VERSION: int
-
-AF_LINK: AddressFamily  # Availability: BSD, macOS
-
-# BDADDR_* and HCI_* listed with other bluetooth constants below
-
-SO_DOMAIN: int
-SO_PASSSEC: int
-SO_PEERSEC: int
-SO_PROTOCOL: int
-TCP_CONGESTION: int
-TCP_USER_TIMEOUT: int
-
-if sys.platform == "linux" and sys.version_info >= (3, 8):
-    AF_QIPCRTR: AddressFamily
-
-# Semi-documented constants
-# (Listed under "Socket families" in the docs, but not "Constants")
-
-if sys.platform == "linux":
-    # Netlink is defined by Linux
-    AF_NETLINK: AddressFamily
-    NETLINK_ARPD: int
-    NETLINK_CRYPTO: int
-    NETLINK_DNRTMSG: int
-    NETLINK_FIREWALL: int
-    NETLINK_IP6_FW: int
-    NETLINK_NFLOG: int
-    NETLINK_ROUTE6: int
-    NETLINK_ROUTE: int
-    NETLINK_SKIP: int
-    NETLINK_TAPBASE: int
-    NETLINK_TCPDIAG: int
-    NETLINK_USERSOCK: int
-    NETLINK_W1: int
-    NETLINK_XFRM: int
-
-if sys.platform != "win32" and sys.platform != "darwin":
-    # Linux and some BSD support is explicit in the docs
-    # Windows and macOS do not support in practice
-    AF_BLUETOOTH: AddressFamily
-    BTPROTO_HCI: int
-    BTPROTO_L2CAP: int
-    BTPROTO_RFCOMM: int
-    BTPROTO_SCO: int  # not in FreeBSD
-
-    BDADDR_ANY: str
-    BDADDR_LOCAL: str
-
-    HCI_FILTER: int  # not in NetBSD or DragonFlyBSD
-    # not in FreeBSD, NetBSD, or DragonFlyBSD
-    HCI_TIME_STAMP: int
-    HCI_DATA_DIR: int
-
-if sys.platform == "darwin":
-    # PF_SYSTEM is defined by macOS
-    PF_SYSTEM: int
-    SYSPROTO_CONTROL: int
+    from _socket import (
+        RCVALL_IPLEVEL as RCVALL_IPLEVEL,
+        RCVALL_MAX as RCVALL_MAX,
+        RCVALL_OFF as RCVALL_OFF,
+        RCVALL_ON as RCVALL_ON,
+        RCVALL_SOCKETLEVELONLY as RCVALL_SOCKETLEVELONLY,
+        SIO_KEEPALIVE_VALS as SIO_KEEPALIVE_VALS,
+        SIO_LOOPBACK_FAST_PATH as SIO_LOOPBACK_FAST_PATH,
+        SIO_RCVALL as SIO_RCVALL,
+    )
+
+_T = TypeVar("_T")
+
+# Re-exported from errno
+EBADF: int
+EAGAIN: int
+EWOULDBLOCK: int
 
-# enum versions of above flags
 class AddressFamily(IntEnum):
     AF_UNIX: int
```

Comment:
This technically needs some more platform dependence. On Mac I only get these:
```
In [29]: socket.AddressFamily.__members__
Out[29]: 
mappingproxy({'AF_UNSPEC': <AddressFamily.AF_UNSPEC: 0>,
              'AF_UNIX': <AddressFamily.AF_UNIX: 1>,
              'AF_INET': <AddressFamily.AF_INET: 2>,
              'AF_SNA': <AddressFamily.AF_SNA: 11>,
              'AF_APPLETALK': <AddressFamily.AF_APPLETALK: 16>,
              'AF_ROUTE': <AddressFamily.AF_ROUTE: 17>,
              'AF_LINK': <AddressFamily.AF_LINK: 18>,
              'AF_IPX': <AddressFamily.AF_IPX: 23>,
              'AF_INET6': <AddressFamily.AF_INET6: 30>,
              'AF_SYSTEM': <AddressFamily.AF_SYSTEM: 32>})
```
I'm fine with leaving this as is though.

Reply:
I suspected that these are not up-to-date anymore. But I'm not keen on looking through this in detail at the moment.